base/base_tests/thread_pool_tests.cpp

```diff
@@ -89,23 +90,18 @@ namespace
 
 UNIT_TEST(ThreadPool_ExecutionTaskTest)
 {
-  std::vector<threads::IRoutine *> tasks;
-  for (int i = 0; i < TASK_COUNT - 1; ++i)
-    tasks.push_back(new CancelTestTask(true));
-  // CancelTastTask::Do method should not be called for last task
-  tasks.push_back(new CancelTestTask(false));
-
   int finishCounter = 0;
   Condition cond;
-  base::thread_pool::routine::ThreadPool pool(4, std::bind(&JoinFinishFunction, std::placeholders::_1,
-                                        std::ref(finishCounter), std::ref(cond)));
+  base::ThreadPool pool(4, std::bind(&JoinFinishFunction, std::placeholders::_1,
+                                     std::ref(finishCounter), std::ref(cond)));
 
-  for (size_t i = 0; i < tasks.size(); ++i)
-    pool.PushBack(tasks[i]);
+  for (int i = 0; i < TASK_COUNT - 1; ++i)
+    pool.PushBack(new CancelTestTask(true));
 
   // CancelTastTask::Do method should not be called for last task
-  tasks.back()->Cancel();
-  tasks.clear();
+  auto * p = new CancelTestTask(false);
```

Comment:
Is it properly deleted? unique_ptr?

Reply:
base::ThreadPool is designed that tasks are deleted by separate finish function.